packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -531,26 +562,143 @@ class CupertinoPopupSurface extends StatelessWidget {
+  /// [CupertinoPopupSurface].
+  ///
+  /// Eyeballed from the iOS 17 simulator.
+  static const double defaultBlurSigma = 30.0;
+
+  /// The default corner radius of a [CupertinoPopupSurface].
+  static const BorderRadius _clipper = BorderRadius.all(Radius.circular(14));
+
+  // The [ColorFilter] matrix used to saturate widgets underlying a
+  // [CupertinoPopupSurface] when the ambient [CupertinoThemeData.brightness] is
+  // [Brightness.light].
+  //
+  // To derive this matrix, the saturation matrix was taken from
+  // https://docs.rainmeter.net/tips/colormatrix-guide/ and was tweaked to
+  // resemble the iOS 17 simulator.
+  //
+  // The matrix can be derived from the following function:
+  // static List<double> get _lightSaturationMatrix {
+  //    const double lightLumR = 0.26;
+  //    const double lightLumG = 0.4;
+  //    const double lightLumB = 0.17;
+  //    const double saturation = 2.0;
+  //    const double sr = (1 - saturation) * lightLumR;
+  //    const double sg = (1 - saturation) * lightLumG;
+  //    const double sb = (1 - saturation) * lightLumB;
+  //    return <double>[
+  //      sr + saturation, sg, sb, 0.0, 0.0,
+  //      sr, sg + saturation, sb, 0.0, 0.0,
+  //      sr, sg, sb + saturation, 0.0, 0.0,
+  //      0.0, 0.0, 0.0, 1.0, 0.0,
+  //    ];
+  //  }
+  static const List<double> _lightSaturationMatrix = <double>[
+     1.74, -0.40, -0.17, 0.00, 0.00,
+    -0.26,  1.60, -0.17, 0.00, 0.00,
+    -0.26, -0.40,  1.83, 0.00, 0.00,
+     0.00,  0.00,  0.00, 1.00, 0.00
+  ];
+
+  // The [ColorFilter] matrix used to saturate widgets underlying a
+  // [CupertinoPopupSurface] when the ambient [CupertinoThemeData.brightness] is
+  // [Brightness.dark].
+  //
+  // To derive this matrix, the saturation matrix was taken from
+  // https://docs.rainmeter.net/tips/colormatrix-guide/ and was tweaked to
+  // resemble the iOS 17 simulator.
+  //
+  // The matrix can be derived from the following function:
+  // static List<double> get _darkSaturationMatrix {
+  //    const double additive = 0.3;
+  //    const double darkLumR = 0.45;
+  //    const double darkLumG = 0.8;
+  //    const double darkLumB = 0.16;
+  //    const double saturation = 1.7;
+  //    const double sr = (1 - saturation) * darkLumR;
+  //    const double sg = (1 - saturation) * darkLumG;
+  //    const double sb = (1 - saturation) * darkLumB;
+  //    return <double>[
+  //      sr + saturation, sg, sb, 0.0, additive,
+  //      sr, sg + saturation, sb, 0.0, additive,
+  //      sr, sg, sb + saturation, 0.0, additive,
+  //      0.0, 0.0, 0.0, 1.0, 0.0,
+  //    ];
+  //  }
+  static const List<double> _darkSaturationMatrix = <double>[
+     1.39, -0.56, -0.11, 0.00, 0.30,
+    -0.32,  1.14, -0.11, 0.00, 0.30,
+    -0.32, -0.56,  1.59, 0.00, 0.30,
+     0.00,  0.00,  0.00, 1.00, 0.00
+  ];
+
+  ImageFilter? _buildFilter(Brightness? brightness) {
+    if ((kIsWeb && !isSkiaWeb) || !debugIsVibrancePainted) {
```

Comment:
Wrap `debugIsVibrancePainted` in an assert so that it only applies in debug mode.
```suggestion
    bool isVibrancePainted = true;
    assert(() {
      isVibrancePainted = debugIsVibrancePainted;
      return true;
    }());
    if ((kIsWeb && !isSkiaWeb) || !isVibrancePainted) {
```

Reply:
Added